external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSerdeSuite.scala

```diff
@@ -49,6 +56,160 @@ class AvroSerdeSuite extends SparkFunSuite {
+      val javaDecimal = grec.get("javaBigDecimal").asInstanceOf[GenericFixed]
+      val sparkDecimal = grec.get("sparkDecimal").asInstanceOf[GenericFixed]
+
+      assert(javaDecimal === sparkDecimal)
+      assert(
+        new DecimalConversion().fromFixed(sparkDecimal, fixedSchema, logicalType) ===
+          new java.math.BigDecimal("1000.12"))
+    }
+  }
+
+  test("Serialize DecimalType to Avro BYTES with logical type decimal") {
+    withFieldMatchType { fieldMatch =>
+      val structType = StructType(
+        Seq(
+          StructField("javaBigDecimal", DecimalType(6, 2), nullable = false),
+          StructField("sparkDecimal", DecimalType(6, 2), nullable = false)))
+
+      val bytesSchema = Schema.create(BYTES)
+      val logicalType = LogicalTypes.decimal(6, 2)
+      val fieldSchema = logicalType.addToSchema(bytesSchema)
+
+      val avroSchema = Schema.createRecord(
+        "name",
+        "doc",
+        "space",
+        true,
+        Seq(
+          new Schema.Field("javaBigDecimal", fieldSchema, "", null.asInstanceOf[AnyVal]),
+          new Schema.Field("sparkDecimal", fieldSchema, "", null.asInstanceOf[AnyVal])).asJava)
+
+      val serializer = Serializer.create(structType, avroSchema, fieldMatch)
+
+      val input = InternalRow(new java.math.BigDecimal("1000.12"), Decimal("1000.12"))
+
+      val grec = serializer.serialize(input).asInstanceOf[GenericRecord]
+      val javaDecimal = grec.get("javaBigDecimal").asInstanceOf[ByteBuffer]
+      val sparkDecimal = grec.get("sparkDecimal").asInstanceOf[ByteBuffer]
+
+      assert(javaDecimal === sparkDecimal)
+      assert(
+        new DecimalConversion().fromBytes(sparkDecimal, bytesSchema, logicalType) ===
+          new java.math.BigDecimal("1000.12"))
+    }
+  }
+
+  test("Serialize DateType to Avro INT") {
+    withFieldMatchType { fieldMatch =>
+      val structType = StructType(
+        Seq(
+          StructField("javaSqlDate", DateType, nullable = false),
+          StructField("java8TimeDate", DateType, nullable = false)))
+
+      val dateSchema = Schema.create(INT)
+
+      val avroSchema = Schema.createRecord(
+        "name",
+        "doc",
+        "space",
+        true,
+        Seq(
+          new Schema.Field("javaSqlDate", dateSchema, "", null.asInstanceOf[AnyVal]),
+          new Schema.Field("java8TimeDate", dateSchema, "", null.asInstanceOf[AnyVal])).asJava)
+
+      val serializer = Serializer.create(structType, avroSchema, fieldMatch)
+
+      val input = InternalRow(
+        new java.sql.Date(1643121231000L),
+        Instant.ofEpochMilli(1643121231000L).atZone(ZoneId.of("UTC")).toLocalDate())
+
+      val grec = serializer.serialize(input).asInstanceOf[GenericRecord]
+      val javaSqlDate = grec.get("javaSqlDate").asInstanceOf[Int]
+      val java8TimeDate = grec.get("java8TimeDate").asInstanceOf[Int]
+
+      assert(javaSqlDate === java8TimeDate)
+      assert(javaSqlDate === 19017) // 19017 is 25 January 2022
+    }
+  }
+
+  test(s"""
```

Comment:
We need to test TimestampNTZ as well

Reply:
good one. found a bug with the added test.

Reply:
There is a bug in the test case